core/qq_group_cache.py

```
@@ -43,11 +51,18 @@ async def list_groups(
         }
 
     def _is_fresh(self) -> bool:
-        return (time.time() - self._last_refresh_at) < self.ttl_seconds
+        now = time.time()
+        if self._available and self._groups:
+            return (now - self._last_refresh_at) < self.ttl_seconds
+        # 失败/空结果：短冷却，避免长时间卡在 unavailable
```

Ruff (0.16.0)

[warning] Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)